libcontainer/cgroups/v1_utils.go

```diff
@@ -43,11 +49,59 @@ func IsNotFound(err error) bool {
 	return ok
 }
 
+func tryDefaultPath(cgroupPath, subsystem string) string {
+	if !strings.HasPrefix(defaultPrefix, cgroupPath) {
+		return ""
+	}
+
+	// remove possible prefix
+	subsystem = strings.TrimPrefix(subsystem, CgroupNamePrefix)
+
+	// Make sure we're still under defaultPrefix, and resolve
+	// a possible symlink (like cpu -> cpu,cpuacct).
+	path, err := securejoin.SecureJoin(defaultPrefix, subsystem)
+	if err != nil {
+		return ""
+	}
+
+	// (1) path should be a directory.
+	st, err := os.Lstat(path)
+	if err != nil || !st.IsDir() {
+		return ""
+	}
+
+	// (2) path should be a mount point.
+	pst, err := os.Lstat(filepath.Dir(path))
+	if err != nil {
+		return ""
+	}
+
+	if st.Sys().(*syscall.Stat_t).Dev == pst.Sys().(*syscall.Stat_t).Dev {
+		// parent dir has the same dev -- path is not a mount point
+		return ""
+	}
+
+	// (3) path should have 'cgroup' fs type.
+	fst := unix.Statfs_t{}
+	err = unix.Statfs(path, &fst)
+	if err != nil || fst.Type != unix.CGROUP_SUPER_MAGIC {
```

Comment:
Do we want to allow FUSE magic as well? (for LXCFS)

Reply:
Well, the code of FindCgroupMountpoint do not have support for lxcfs either, so it would be useless to add it here.

Besides, it seems like this feature is obsoleted (by cgroupns). From https://github.com/lxc/lxcfs

> Prior to the implementation of cgroup namespaces by Serge Hallyn LXCFS also provided a container aware cgroupfs tree. It took care that the container only had access to cgroups underneath it's own cgroups and thus provided additional safety. For systems without support for cgroup namespaces LXCFS will still provide this feature but it is mostly considered deprecated.

Reply:
> Well, the code of FindCgroupMountpoint do not have support for lxcfs either, so it would be useless to add it here

Unless, of course, LXCFS shadows /proc/self/mountinfo as well, in this case we might need to support it here as well.

Summoning @hallyn @brauner @stgraber -- what do you guys think?

Reply:
lxcfs cannot shadow /proc/self/* because those entries show up dynamically or are magic links so a bit incompatible with bind-mounts.

The fuse cgroupfs view is currently mostly unused thanks to cgns doing a much cleaner job but it may make a come back as we need a solution for running containers that are only cgroupv1 aware on cgroupv2-only hosts and there hasn't been much interest for an in-kernel solution to that...

Reply:
> we need a solution for running containers that are only cgroupv1 aware on cgroupv2-only hosts

This is hardly a runc use case.

Anyway, since supporting lxcfs requires more changes than just modifying this code, it is probably out of scope for this PR and can be implemented separately later.